scr/main/java/bittercode/service/impl/BookServiceImpl.java

```diff
@@ -0,0 +1,193 @@
+package com.bittercode.service.impl;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.bittercode.constant.ResponseCode;
+import com.bittercode.constant.db.BooksDBConstants;
+import com.bittercode.model.Book;
+import com.bittercode.model.StoreException;
+import com.bittercode.service.BookService;
+import com.bittercode.util.DBUtil;
+
+public class BookServiceImpl implements BookService {
+
+    private static final String getAllBooksQuery = "SELECT * FROM " + BooksDBConstants.TABLE_BOOK;
+    private static final String getBookByIdQuery = "SELECT * FROM " + BooksDBConstants.TABLE_BOOK
+            + " WHERE " + BooksDBConstants.COLUMN_BARCODE + " = ?";
+
+    private static final String deleteBookByIdQuery = "DELETE FROM " + BooksDBConstants.TABLE_BOOK + "  WHERE "
+            + BooksDBConstants.COLUMN_BARCODE + "=?";
+
+    private static final String addBookQuery = "INSERT INTO " + BooksDBConstants.TABLE_BOOK + "  VALUES(?,?,?,?,?)";
+
+    private static final String updateBookQtyByIdQuery = "UPDATE " + BooksDBConstants.TABLE_BOOK + " SET "
+            + BooksDBConstants.COLUMN_QUANTITY + "=? WHERE " + BooksDBConstants.COLUMN_BARCODE
+            + "=?";
+
+    private static final String updateBookByIdQuery = "UPDATE " + BooksDBConstants.TABLE_BOOK + " SET "
+            + BooksDBConstants.COLUMN_NAME + "=? , "
+            + BooksDBConstants.COLUMN_AUTHOR + "=?, "
+            + BooksDBConstants.COLUMN_PRICE + "=?, "
+            + BooksDBConstants.COLUMN_QUANTITY + "=? "
+            + "  WHERE " + BooksDBConstants.COLUMN_BARCODE
+            + "=?";
+
+    @Override
+    public Book getBookById(String bookId) throws StoreException {
+        Book book = null;
+        Connection con = DBUtil.getConnection();
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/PrajaktaBendreBito/BookStore/pull/1#issuecomment-2438181340>#16fdcd</a><span> - 10/25/2024, 04:08 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> Database connection is created for each method call, which can lead to connection pool exhaustion and performance degradation.
</div>
<div id="fix">
<b>Fix:</b> Implement a connection pool and pass the connection as a parameter to the method or use a try-with-resources statement to ensure proper connection management.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -40,8 +40,7 @@
  @Override
  public Book getBookById(String bookId) throws StoreException {
      Book book = null;
 -    Connection con = DBUtil.getConnection();
 -    try {
 +    try (Connection con = DBUtil.getConnection()) {
          PreparedStatement ps = con.prepareStatement(getBookByIdQuery);
          ps.setString(1, bookId);
          ResultSet rs = ps.executeQuery();
 @@ -55,8 +54,8 @@

              book = new Book(bCode, bName, bAuthor, bPrice, bQty);
          }
 -    } catch (SQLException e) {
 -
 +    } catch (SQLException | StoreException e) {
 +        throw new StoreException("Error retrieving book: " + e.getMessage(), e);
      }
      return book;
  }
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged